### DISCUSSION TITLE
<Modal /> should unmount itself from the dom when closed

### DISCUSSION THREAD

**Discussion by sep2** [mantinedev/mantine#1919] [Feedback]
Now there is an empty div in the dom even the modal itself is closed. In my case I have a datagrid where each cell is a button that can open a modal when click. only a few of them will be opened at a time, others will be in closed state most of the time. In the current modal implementation If I have hundreads of those cells it will generate hundreds of empty divs in the dom for nothing.

```
<div><div dir="ltr"></div></div>
```

**Comment by rtivital** [comment]
I advise not to render Modal for each cell, add one Modal component that will handle all cells.

**Reply by sep2** [reply]
My case is more complicated that a cell modal can itself open another cell modal overlapping the previous one. So I can't use a single modal component to implement this.
